### PR TITLE
Mark various interfaces as api

### LINKED ISSUE #32020: Mark Msrp, Newsletter, PageCache, Payment, Paypal and Quote interfaces as API
The following interfaces should be marked as API in 2.4-develop branch

1. app/code/Magento/Msrp/Pricing/MsrpPriceCalculatorInterface.php
1. app/code/Magento/Newsletter/Model/SubscriptionManagerInterface.php
1. app/code/Magento/PageCache/Model/Spi/PageCacheTagsPreprocessorInterface.php
1. app/code/Magento/Payment/Api/Data/PaymentAdditionalInfoInterface.php
1. app/code/Magento/Payment/Model/Method/Online/GatewayInterface.php
1. app/code/Magento/Payment/Model/Method/TransparentInterface.php
1. app/code/Magento/Paypal/Helper/Shortcut/ValidatorInterface.php
1. app/code/Magento/Paypal/Model/Billing/Agreement/MethodInterface.php
1. app/code/Magento/Paypal/Model/IpnInterface.php
1. app/code/Magento/Paypal/Model/Payflow/Service/Response/Handler/HandlerInterface.php
1. app/code/Magento/Paypal/Model/Payflow/Service/Response/ValidatorInterface.php
1. app/code/Magento/Quote/Api/ChangeQuoteControlInterface.php
1. app/code/Magento/Quote/Model/Cart/BuyRequest/BuyRequestDataProviderInterface.php
1. app/code/Magento/Quote/Model/Product/QuoteItemsCleanerInterface.php
1. app/code/Magento/Quote/Model/Quote/Address/CustomAttributeListInterface.php
1. app/code/Magento/Quote/Model/Quote/Address/Total/CollectorInterface.php
1. app/code/Magento/Quote/Model/Quote/Address/Total/ReaderInterface.php
1. app/code/Magento/Quote/Model/Quote/Item/CartItemProcessorInterface.php
1. app/code/Magento/Quote/Model/ValidationRules/QuoteValidationRuleInterface.php
1. app/code/Magento/QuoteGraphQl/Model/Cart/BuyRequest/BuyRequestDataProviderInterface.php
1. app/code/Magento/QuoteGraphQl/Model/Cart/Payment/AdditionalDataProviderInterface.php
1. app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressesOnCartInterface.php
1. app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingMethodsOnCartInterface.php
1. app/code/Magento/QuoteGraphQl/Model/CartItem/DataProvider/CustomizableOptionValueInterface.php
1. app/code/Magento/QuoteGraphQl/Model/CartItem/DataProvider/Processor/ItemDataProcessorInterface.php

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

The following interfaces marked as API

app/code/Magento/Msrp/Pricing/MsrpPriceCalculatorInterface.php
app/code/Magento/Newsletter/Model/SubscriptionManagerInterface.php
app/code/Magento/PageCache/Model/Spi/PageCacheTagsPreprocessorInterface.php
app/code/Magento/Payment/Api/Data/PaymentAdditionalInfoInterface.php

app/code/Magento/Payment/Model/Method/Online/GatewayInterface.php <b>(non-api)</b> because it referenced to deprecated argument interface
Magento\Payment\Model\Method\ConfigInterface. We should replace this with stable api such as Magento\Payment\Gateway\ConfigInterface

app/code/Magento/Payment/Model/Method/TransparentInterface.php
app/code/Magento/Paypal/Helper/Shortcut/ValidatorInterface.php
app/code/Magento/Paypal/Model/Billing/Agreement/MethodInterface.php
app/code/Magento/Paypal/Model/IpnInterface.php
app/code/Magento/Paypal/Model/Payflow/Service/Response/Handler/HandlerInterface.php

app/code/Magento/Paypal/Model/Payflow/Service/Response/ValidatorInterface.php <b>(non-api)</b> references following non-public types:
Magento\Paypal\Model\Payflow\Transparent

app/code/Magento/Quote/Api/ChangeQuoteControlInterface.php
app/code/Magento/Quote/Model/Cart/BuyRequest/BuyRequestDataProviderInterface.php
app/code/Magento/Quote/Model/Product/QuoteItemsCleanerInterface.php
app/code/Magento/Quote/Model/Quote/Address/CustomAttributeListInterface.php
app/code/Magento/Quote/Model/Quote/Address/Total/CollectorInterface.php
app/code/Magento/Quote/Model/Quote/Address/Total/ReaderInterface.php
app/code/Magento/Quote/Model/Quote/Item/CartItemProcessorInterface.php
app/code/Magento/Quote/Model/ValidationRules/QuoteValidationRuleInterface.php
app/code/Magento/QuoteGraphQl/Model/Cart/BuyRequest/BuyRequestDataProviderInterface.php
app/code/Magento/QuoteGraphQl/Model/Cart/Payment/AdditionalDataProviderInterface.php
app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressesOnCartInterface.php
app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingMethodsOnCartInterface.php
app/code/Magento/QuoteGraphQl/Model/CartItem/DataProvider/CustomizableOptionValueInterface.php
app/code/Magento/QuoteGraphQl/Model/CartItem/DataProvider/Processor/ItemDataProcessorInterface.php

Marked also references class/interface as api

app/code/Magento/Paypal/Model/Billing/AbstractAgreement.php

app/code/Magento/Quote/Model/Cart/Data/CartItem.php
app/code/Magento/Quote/Model/Quote/Item/Option.php

app/code/Magento/GraphQl/Model/Query/ContextInterface.php

lib/internal/Magento/Framework/GraphQl/Query/Resolver/ContextInterface.php
lib/internal/Magento/Framework/GraphQl/Exception/GraphQlNoSuchEntityException.php
lib/internal/Magento/Framework/GraphQl/Exception/GraphQlInputException.php
lib/internal/Magento/Framework/GraphQl/Exception/GraphQlAuthorizationException.php
lib/internal/Magento/Framework/GraphQl/Exception/GraphQlAuthenticationException.php

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32020

### Manual testing scenarios (*)

1. ...
2. ...

### Questions or comments

CC: @sivaschenko can you check valid mark api for app/code/Magento/Payment/Model/Method/Online/GatewayInterface.php
See more details above in PR description section

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by BarnyShergold** [comment]
@magento run all tests

**Comment by BarnyShergold** [comment]
@magento run Functional Tests B2B

**Comment by BarnyShergold** [comment]
@magento run Static Tests

**Comment by BarnyShergold** [comment]
@magento run Database Compare

**Comment by BarnyShergold** [comment]
@magento run Static Tests

**Comment by BarnyShergold** [comment]
@magento run Database Compare

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
The Database Compare & Static Tests keep failing but I can't see why what you've added would cause these issues. Can you check?

**Comment by mrtuvn** [comment]
Seem false positive change

**Comment by mrtuvn** [comment]
@magento run Static Tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests EE

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Review by BarnyShergold — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @BarnyShergold, thank you for the review.
[ENGCOM-8800](https://jira.corp.magento.com/browse/ENGCOM-8800) has been created to process this Pull Request

**Comment by mrtuvn** [comment]
Fix static
@magento run all tests

**Comment by mrtuvn** [comment]
Fix static
@magento run all tests

**Comment by mrtuvn** [comment]
Fix static
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests, Semantic Version Checker, Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run Semantic Version Checker

File: app/code/Magento/Payment/Model/Method/Online/GatewayInterface.php [hunk 1]
```diff
@@ -6,12 +6,12 @@
...
+use Magento\Payment\Gateway\ConfigInterface;
 
 /**
  * Gateway interface for online payment methods
  *
- * @author     Magento Core Team <core@magentocommerce.com>
+ * @api
  */
 interface GatewayInterface
 {
```
Comment (mrtuvn) [inline app/code/Magento/Payment/Model/Method/Online/GatewayInterface.php]: Replace deprecated interface with interface marked @api

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Static Tests, Unit Tests

**Comment by mrtuvn** [comment]
@magento run Static Tests

**Comment by mrtuvn** [comment]
@magento run WebAPI Tests, Unit Tests, Semantic Version Checker, Integration Tests,Functional Tests EE, Functional Tests CE, Functional Tests B2B, Database Compare,Magento Health Index, Sample Data Tests B2B, Sample Data Tests CE, Sample Data Tests EE

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
Re-targeted PR branch to 2.4-develop as internal discussed with @gabrieldagama

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests, Functional Tests CE, Functional Tests EE

**Review by gabrieldagama — APPROVED** [review]

**Comment by gabrieldagama** [comment]
@magento import code to https://github.com/magento-engcom/magento2ce

**Comment by magento-engcom-team** [comment]
@gabrieldagama the branch with code successfully imported into`magento-engcom/magento2ce` repository. Branch name: `imported-magento-magento2-32157`.

**Pull request merged**